aws/resource_aws_launch_configuration.go

```diff
@@ -526,11 +527,26 @@ func resourceAwsLaunchConfigurationRead(d *schema.ResourceData, meta interface{}
 	d.Set("ebs_optimized", lc.EbsOptimized)
 	d.Set("spot_price", lc.SpotPrice)
 	d.Set("enable_monitoring", lc.InstanceMonitoring.Enabled)
-	d.Set("security_groups", lc.SecurityGroups)
 	d.Set("associate_public_ip_address", lc.AssociatePublicIpAddress)
+	if len(lc.SecurityGroups) > 0 {
+		securityGroups := make([]string, 0, len(lc.SecurityGroups))
+		for _, sg := range lc.SecurityGroups {
+			securityGroups = append(securityGroups, *sg)
+		}
+		d.Set("security_groups", securityGroups)
+	}
+	if *lc.UserData != "" {
```

Comment:
We should safely dereference this pointer or it can cause a panic: `if aws.StringValue(lc.UserData) !=""`